src/runner/UpdateUtils.cpp

```diff
@@ -276,6 +278,8 @@ void CheckForUpdatesCallback()
         }
         
         ProcessNewVersionInfo(*new_version_info, state, download_update, false);
+
+        std::lock_guard<std::mutex> lock(state_mutex);
```

Comment:
`UpdateState::store` already uses mutex internally, why do we need another one? 🤔

Reply:
Accidentally committed. Changes to the file have been undone.